### PR TITLE
popura: init at 0.4.6

### LINKED ISSUE #163221: Popura ポプラ: alternative Yggdrasil network client
**Project description**
Popura, an alternative Yggdrasil network client

Yggdrasil Network is a peer-to-peer IPv6 network with link-local peer discovery, automatic end-to-end encryption, distributed IP address allocation, and DHT-based routing information exchange.

Popura uses the same Yggdrasil core API internally, but adds some useful experimental features which the original client lacks.

By default, it works just like the original yggdrasil client, all features must be enabled manually. Popura adds new command line flags and config file sections to control those features.
Features

-   Autopeering over the Internet

-  Built-in decentralized DNS system (meshname)

**Metadata**

* homepage URL: https://github.com/popura-network/Popura
* source URL: https://github.com/popura-network/Popura
* license: lgpl3
* platforms: linux

### PR DESCRIPTION
**Pull request by urandom2**
###### Description of changes

Fixes #163221

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/tools/networking/popura/default.nix [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-iCu6/vD4vgn7aGdwK+OB8ib/QwUwoFuxDUs7vqbTZQc=";
+  };
+
+  vendorHash = "sha256-9lQC35yt1S2uch3qgwNfa/1FHy+Qi1D5Jo7DWNMgU9w=";
+
+  meta = with lib; {
+    description = "Popura ポプラ: alternative Yggdrasil network client";
+    homepage = "https://popura-network.github.io";
```
Comment (figsoda) [inline pkgs/tools/networking/popura/default.nix:21]: the original link doesn't work for me
```suggestion
    homepage = "https://github.com/popura-network/popura";
```

File: pkgs/tools/networking/popura/default.nix [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+    license = licenses.lgpl3;
```
Comment (figsoda) [inline pkgs/tools/networking/popura/default.nix:22]:
```suggestion
    license = licenses.lgpl3Only;
```

File: pkgs/tools/networking/popura/default.nix [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+    owner = "${pname}-network";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-iCu6/vD4vgn7aGdwK+OB8ib/QwUwoFuxDUs7vqbTZQc=";
+  };
+
+  vendorHash = "sha256-9lQC35yt1S2uch3qgwNfa/1FHy+Qi1D5Jo7DWNMgU9w=";
+
+  meta = with lib; {
+    description = "Popura ポプラ: alternative Yggdrasil network client";
```
Comment (figsoda) [inline pkgs/tools/networking/popura/default.nix:20]: > [For brevity, don’t repeat the name of package](https://nixos.org/manual/nixpkgs/unstable/#var-meta-description)
```suggestion
    description = "An alternative Yggdrasil network client";
```

File: pkgs/tools/networking/popura/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{ buildGoModule
+, fetchFromGitHub
+, lib
```
Comment (figsoda) [inline pkgs/tools/networking/popura/default.nix:3]: > [Arguments should be listed in the order they are used, with the exception of lib, which always goes first.](https://nixos.org/manual/nixpkgs/unstable/#sec-syntax)
```suggestion
{ lib
, buildGoModule
, fetchFromGitHub
```

File: pkgs/tools/networking/popura/default.nix [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+
+  src = fetchFromGitHub {
+    owner = "${pname}-network";
+    repo = pname;
+    rev = "v${version}";
+    hash = "sha256-iCu6/vD4vgn7aGdwK+OB8ib/QwUwoFuxDUs7vqbTZQc=";
+  };
+
+  vendorHash = "sha256-9lQC35yt1S2uch3qgwNfa/1FHy+Qi1D5Jo7DWNMgU9w=";
+
```
Comment (figsoda) [inline pkgs/tools/networking/popura/default.nix:18]:
```suggestion

  ldflags = let pkgSrc = "github.com/yggdrasil-network/yggdrasil-go/src/version"; in [
    "-s"
    "-w"
    "-X=${pkgSrc}.buildName=yggdrasil"
    "-X=${pkgSrc}.buildVersion=${version}"
  ];

```

File: pkgs/tools/networking/popura/default.nix [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+  meta = with lib; {
+    description = "Popura ポプラ: alternative Yggdrasil network client";
+    homepage = "https://popura-network.github.io";
+    license = licenses.lgpl3;
+    maintainers = with maintainers; [ urandom ];
```
Comment (figsoda) [inline pkgs/tools/networking/popura/default.nix:23]:
```suggestion
    maintainers = with maintainers; [ urandom ];
    mainProgram = "yggdrasil";
```

**Review by figsoda — CHANGES_REQUESTED** [review]

File: pkgs/tools/networking/popura/default.nix (hunk 1 above)
Comment (urandom2) [reply pkgs/tools/networking/popura/default.nix:21]: hmm, it does look broken; sad

**Review by figsoda — APPROVED** [review]

**Pull request merged**